### PR TITLE
Add smem package

### PR DESCRIPTION
**Pull request by cstrouse**
Smem is a memory profiling tool that uses proportial set sizes to give a more
useful measurement of memory usage in virtual memory systems.

Tested as working on XE500C13-K01US.

### CODE REVIEW

**Comment by uberhacker** [comment]
Thank you for the new package @cstrouse.  Please see the comments under `Files changed`.

**Comment by cstrouse** [comment]
@uberhacker I don't see any comments there.

**Comment by uberhacker** [comment]
Weird.  OK, I'll post them here:
Does this install with python3? If so, we might want something like this instead:
```
depends_on 'python27' unless File.exists? "#{CREW_PREFIX}/bin/python"
```
No `configure` script?

File: packages/smem.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  source_url 'https://www.selenic.com/smem/download/smem-1.4.tar.gz' 
+  source_sha256 '2ea9f878f4cf3c276774c3f7e2a41977a1f2d64f98d2dcb6a15f1f3d84df61ec'
+
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
+
+  depends_on 'buildessential' => :build
+  depends_on 'python27'
```
[the remark this answers is not in the document]
Comment (cstrouse) [reply packages/smem.rb]: I don't think it works with python3.

File: packages/smem.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  def self.build
+    system 'make smemcap'
```
[the remark this answers is not in the document]
Comment (cstrouse) [reply packages/smem.rb]: Correct.

**Pull request merged**